src/network/url_downloader.cpp

```diff
@@ -99,21 +99,57 @@ QByteArray download(QNetworkAccessManager* manager, const Time& timeout, QUrl co
 
         const auto msg = reply->errorString().toStdString();
 
+        if (reply->error() == QNetworkReply::ProxyAuthenticationRequiredError)
+            reply->abort();
+
         if (abort_download)
             throw mp::AbortedDownloadException{msg};
         else
             throw mp::DownloadException{url.toString().toStdString(), download_timeout.isActive() ? msg : "Network timeout"};
     }
     return reply->readAll();
 }
+
+std::unique_ptr<QNetworkProxy> discover_http_proxy()
+{
+    std::unique_ptr<QNetworkProxy> proxy_ptr{nullptr};
+
+    QString http_proxy{qgetenv("http_proxy")};
+    if (http_proxy.isEmpty())
+    {
+        // Some OS's are case senstive
+        http_proxy = qgetenv("HTTP_PROXY");
+    }
+
+    if (!http_proxy.isEmpty())
+    {
+        if (!http_proxy.startsWith("http://"))
+        {
+            http_proxy.prepend("http://");
```

Comment:
SOCKS support can come later.  If a system level SOCKS proxy is set, Qt will automatically pick it up anyways.

Reply:
My point is that passing `http://socks://blah` to Qt doesn't seem right, even if we don't have specific support for SOCKS. Perhaps this could be checked for a `"://"` sub-string instead. Otherwise, maybe it is preferable to require the protocol to be part of the env var.

Reply:
An env var defined as `http_proxy=socks://blah` is not a valid proxy and would break things regardless of how we are parsing this.  This is for doing something like the following:

`http_proxy=192.168.1.3:3128` and then adding `http://` to the beginning of the IP address (or host name if it's given) so we can then pass a valid URL to `QUrl`.

Reply:
I am not sure I follow, `http_proxy=socks5://something` is normally correct, does Qt not like that?

Reply:
Ah, ok, I see what you mean.  Sure, I should just check to make sure like you mentioned previously.